### PR TITLE
OffRouteDetector Check for Upcoming Step

### PR DESCRIPTION
**Pull request by danesfeder**
When a user is determined "off-route" but they are within an acceptable radius of the upcoming step, do not send an off-route

### CODE REVIEW

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java [hunk 1]
```diff
@@ -24,19 +25,22 @@
   @Override
   public boolean isUserOffRoute(Location location, RouteProgress routeProgress, MapboxNavigationOptions options) {
     Position futurePosition = getFuturePosition(location, options);
-    double radius = Math.min(options.maximumDistanceOffRoute(),
+    double radius = Math.max(options.maximumDistanceOffRoute(),
```
Comment (cammace) [inline navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java]: nice catch, matches iOS now

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java [hunk 2]
```diff
@@ -24,19 +25,22 @@
...
     // the exit bearing.
     boolean isCloseToUpcomingStep;
 
-    if (routeProgress.currentLegProgress().upComingStep() != null) {
-      isCloseToUpcomingStep = userTrueDistanceFromRoute(futurePosition, routeProgress) < radius;
+    LegStep upComingStep = routeProgress.currentLegProgress().upComingStep();
+    if (upComingStep != null) {
+      isCloseToUpcomingStep = userTrueDistanceFromStep(futurePosition, upComingStep) < radius;
       if (isOffRoute && isCloseToUpcomingStep) {
+        // TODO increment step index
```
Comment (cammace) [inline navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java]: Should be able to pass the `indices` reference and use `NavigationHelper.increaseIndices()` to increase.

Comment (danesfeder) [reply navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java]: Sweet - @ericrwolfe do you want to do this now? Or should I ticket

Comment (ericrwolfe) [reply navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java]: Be mindful that the name of this method `isUserOffRoute()` implies it shouldn't have any side effects, particularly if this were to be called somewhere else.

Probably best to move the `isCloseToUpcomingStep` check and the `increaseIndices()` call out into `handleRequest` like we talked about.

**Pull request merged**